### PR TITLE
nerdctl compose up

### PR DESCRIPTION
**Pull request by AkihiroSuda**
e.g.,
```console
$ nerdctl compose -f ./examples/compose-wordpress/docker-compose.yaml up
```

YAML:
```yaml
version: '3.1'

services:

  wordpress:
    image: wordpress:5.7
    restart: always
    ports:
      - 8080:80
    environment:
      WORDPRESS_DB_HOST: db
      WORDPRESS_DB_USER: exampleuser
      WORDPRESS_DB_PASSWORD: examplepass
      WORDPRESS_DB_NAME: exampledb
    volumes:
      - wordpress:/var/www/html

  db:
    image: mariadb:10.5
    restart: always
    environment:
      MYSQL_DATABASE: exampledb
      MYSQL_USER: exampleuser
      MYSQL_PASSWORD: examplepass
      MYSQL_RANDOM_ROOT_PASSWORD: '1'
    volumes:
      - db:/var/lib/mysql

volumes:
  wordpress:
  db:
```

image

- - -

`nerdctl compose up --build` is not implemented yet. Will be implemented in a separate PR later.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -268,3 +276,20 @@ func bashCompleteSnapshotterNames(clicontext *cli.Context) {
 		fmt.Fprintln(clicontext.App.Writer, name)
 	}
 }
+
+func globalFlags(clicontext *cli.Context) (string, []string) {
+	args0 := os.Args[0]
+	if runtime.GOOS == "linux" {
+		var err error
+		args0, err = os.Readlink("/proc/self/exe")
```
Comment (fuweid) [inline main.go:284]: you can try to use `os.Executable()` here :)  it can be used for windows

File: pkg/composer/up_service.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+			}
+		}
+	}
+
+	if uo.Detach {
+		return nil
+	}
+
+	logrus.Info("Attaching to logs")
+	for id, state := range containerStates {
```
Comment (fuweid) [inline pkg/composer/up_service.go:74]: I try it in my local.

One terminal A is `_output/nerdctl compose -f ./examples/compose-wordpress/docker-compose.yaml up` and wait for seconds. Open other terminal B is for `_output/nerdctl compose -f ./examples/compose-wordpress/docker-compose.yaml down` and the terminal A still attach logs.

Comment (fuweid) [reply pkg/composer/up_service.go:74]: maybe we can use `wait` action for watching the container. it can be follow-up.

**Review by fuweid — COMMENTED** [review]
LGTM in general. Thanks!

File: main.go (hunk 1 above)
Comment (AkihiroSuda) [reply main.go]: fixed

File: pkg/composer/up_service.go (hunk 2 above)
Comment (AkihiroSuda) [reply pkg/composer/up_service.go:96]: fixed. We don't have `nerdctl wait <CONTAINER>` yet, so wait for exit of `nerdctl logs -f <CONTAINER>`.

**Comment by AkihiroSuda** [comment]
Thanks for reviewing @fuweid , updated

**Review by ktock — APPROVED** [review]

**Review by fuweid — APPROVED** [review]
LGTM

**Pull request merged**